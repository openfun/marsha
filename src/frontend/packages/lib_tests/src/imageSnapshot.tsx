import path from 'path';
import { cleanup, render } from '@testing-library/react';
import { Grommet } from 'grommet';
import { generateImage, GenerateImageOptions } from 'jsdom-screenshot';
import { GlobalStyles, theme } from 'lib-common';
import React from 'react';

import { wrapInIntlProvider } from './intl';
import { wrapInRouter } from './router';

const checkSnapshot = () => {
  // check if test name contains '[screenshot]'
  if (!expect.getState().currentTestName?.includes('[screenshot]')) {
    throw new Error('[screenshot] is missing from test name');
  }
};

// To disable puppeteer sandbox if needed, launch the test suite
// using DISABLE_PUPPETEER_SANDBOX=1 as environment variable.
// see https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md#setting-up-chrome-linux-sandbox
const disableSandbox = !!process.env.DISABLE_PUPPETEER_SANDBOX;

export const imageSnapshot = async (width?: number, height?: number) => {
  checkSnapshot();
  width = width || 800;
  height = height || 600;
  const generateImageOptions: GenerateImageOptions = {
    viewport: { width, height },
  };
  if (disableSandbox) {
    generateImageOptions.launch = {
      args: ['--no-sandbox', '--disable-setuid-sandbox'],
    };
  }
  const screenshot = await generateImage(generateImageOptions);
  expect(screenshot).toMatchImageSnapshot({
    customDiffDir: path.resolve(__dirname, '../../__diff_output__'),
    failureThreshold: 0.01,
    failureThresholdType: 'percent',
  });
};

export const renderImageSnapshot = async (
  component: JSX.Element,
  width?: number,
  height?: number,
) => {
  checkSnapshot();
  cleanup();
  render(
    wrapInIntlProvider(
      wrapInRouter(
        <Grommet theme={theme} style={{ height: '100%' }}>
          {component}
          <GlobalStyles />
        </Grommet>,
      ),
    ),
  );
  await imageSnapshot(width, height);
};

export const renderIconSnapshot = async (component: JSX.Element) => {
  checkSnapshot();
  await renderImageSnapshot(component, 100, 100);
};
