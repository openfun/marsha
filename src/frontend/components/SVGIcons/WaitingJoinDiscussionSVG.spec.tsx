import React from 'react';
import { renderIconSnapshot } from 'utils/tests/imageSnapshot';

import { WaitingJoinDiscussionSVG } from './WaitingJoinDiscussionSVG';

it('renders WaitingJoinDiscussionSVG correctly', async () => {
  await renderIconSnapshot(<WaitingJoinDiscussionSVG iconColor="#035ccd" />);
});
