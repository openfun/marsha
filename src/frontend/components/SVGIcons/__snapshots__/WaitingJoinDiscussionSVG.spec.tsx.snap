// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders WaitingJoinDiscussionSVG correctly 1`] = `
.c0 path {
  fill: #035ccd;
}

<svg
  className="c0"
  viewBox="0 0 54 54"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    fillRule="evenodd"
    stroke="none"
    strokeWidth="1"
  >
    <g
      transform="translate(2.5 2.5)"
    >
      <path
        d="M16.261 20.33a.817.817 0 000 1.17 4.953 4.953 0 01.462 6.466c-.247.33-.198.775.083 1.072a.834.834 0 001.253-.082 6.63 6.63 0 00-.066-8L30.561 8.39a2.06 2.06 0 012.919 0 2.06 2.06 0 010 2.92l-8.46 8.46a.817.817 0 000 1.171c.329.33.84.33 1.17 0l10.787-10.786a2.06 2.06 0 012.919 0 2.06 2.06 0 010 2.919L29.109 23.859a.817.817 0 000 1.171c.33.33.842.33 1.171 0l9.039-9.038a2.06 2.06 0 012.919 0 2.06 2.06 0 010 2.92L32.028 29.12a.817.817 0 000 1.17c.33.33.842.33 1.172 0l6.712-6.712a2.06 2.06 0 012.92 0 2.06 2.06 0 010 2.92L31.17 38.158c-5.31 5.31-13.937 5.31-19.247 0-5.311-5.31-5.311-13.937 0-19.248l7.586-7.586a2.06 2.06 0 012.92 0 2.06 2.06 0 010 2.919L16.26 20.33zm1.534-13.195c0-.676-.56-1.237-1.237-1.237-.066 0-.115 0-.181.017a9.92 9.92 0 00-8.379 8.378c0 .05-.016.116-.016.182 0 .676.56 1.237 1.237 1.237.626 0 1.138-.462 1.22-1.056a7.417 7.417 0 016.284-6.284 1.258 1.258 0 001.072-1.237zm16.493 33.646c0 .676.561 1.237 1.237 1.237.066 0 .116 0 .182-.016a9.92 9.92 0 008.378-8.379c0-.05.017-.115.017-.181 0-.676-.561-1.237-1.237-1.237-.627 0-1.138.462-1.22 1.055a7.417 7.417 0 01-6.285 6.284c-.61.1-1.072.61-1.072 1.237z"
        fillRule="nonzero"
      />
    </g>
  </g>
</svg>
`;
